### PR TITLE
CLOUDP-280744: Build a framework to execute L1 commands

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes

Added a framework to execute the generated L1 commands.

The goal of this framework (covered by this ticket) is to:
1. take command definition + execution context
2. return the API endpoint as is

_Jira ticket:_ CLOUDP-280744

## Example
```bash
❯ ./bin/atlas api clusters listClustersForAllProjects | jq
{
  "links": [
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/clusters?pageNum=1&itemsPerPage=100",
      "rel": "self"
    }
  ],
  "results": [
    {
      "clusters": [
        ...
      ],
      "groupId": "...",
      "groupName": "Atlas CLI E2E",
      "orgId": "...",
      "orgName": "Atlas CLI E2E",
      "planType": "Atlas",
      "tags": []
    },
    ...
  ],
  "totalCount": 2
}
```

### CODE REVIEW

File: internal/cli/api/api.go [hunk 1]
```diff
@@ -72,16 +75,41 @@ func convertAPIToCobraCommand(command api.Command) (*cobra.Command, error) {
...
+			}
+
+			defer result.Output.Close()
+			_, err = io.Copy(os.Stdout, result.Output)
+			if err != nil {
+				return err
+			}
+
+			if !result.IsSuccess {
+				return errors.New("")
```
Comment (fmenezes) [inline internal/cli/api/api.go:110]: what is this error?

Comment (jeroenvervaeke) [reply internal/cli/api/api.go:110]: I'll improve this

Comment (jeroenvervaeke) [reply internal/cli/api/api.go:112]: Added comments, this code was demo code. It'll be removed and replaced as part of CLOUDP-280745

File: internal/cli/api/api.go [hunk 2]
```diff
@@ -72,16 +75,43 @@ func convertAPIToCobraCommand(command api.Command) (*cobra.Command, error) {
...
 
 	cmd := &cobra.Command{
-		Use:    commandName,
-		Short:  shortDescription,
-		Long:   longDescription,
-		Hidden: true, // TODO: part of CLOUDP-280743 Polish autogenerated docs
+		Use:   commandName,
+		Short: shortDescription,
+		Long:  longDescription,
 		RunE: func(cmd *cobra.Command, _ []string) error {
```
Comment (jeroenvervaeke) [inline internal/cli/api/api.go:81]: NOTE: everything in `RunE` is temporary code. Just to give an idea what the code in CLOUDP-280745 will do. All code in here will be replaced

**Review by fmenezes — APPROVED** [review]

**Pull request merged**